File: libnet/src/main/java/com/uncle2000/libnet/CallBack.java
package com.uncle2000.libnet;

/**
 * Created by danger on 16/6/18.
 */

public interface CallBack<T> {

    default void onPreExecute(T t) {
    }

    void onSuccess(T t);

    void onFail(String message, Throwable t);

}

File: libbase/src/main/java/com/uncle2000/libbase/libnet/CallBack.kt
package com.uncle2000.libbase.libnet

/**
 * Created by danger on 16/6/18.
 */

interface CallBack<T> {

    fun onPreExecute(t: T) {}

    fun onSuccess(t: T?)

    fun onFail(message: String?, t: Throwable?)

}
